modules/carbonAnalyticsAdapter.js

```diff
@@ -0,0 +1,238 @@
+import { logError, generateUUID } from '../src/utils.js';
+import { ajax } from '../src/ajax.js';
+import { getStorageManager } from '../src/storageManager.js';
+import {getGlobal} from '../src/prebidGlobal.js';
+import adapterManager from '../src/adapterManager.js';
+import adapter from '../libraries/analyticsAdapter/AnalyticsAdapter.js';
+import CONSTANTS from '../src/constants.json';
+
+const CARBON_GVL_ID = 493;
+const ANALYTICS_VERSION = 'v1.0';
+const PROFILE_ID_KEY = 'carbon_ccuid';
+const PROFILE_ID_COOKIE = 'ccuid';
+const SESSION_ID_COOKIE = 'ccsid';
+const MODULE_NAME = 'carbon';
+const ANALYTICS_TYPE = 'endpoint';
+
+const DAY_MS = 24 * 60 * 60 * 1000;
+const MINUTE_MS = 60 * 1000;
+
+export const storage = getStorageManager({gvlid: CARBON_GVL_ID, moduleName: 'carbon'});
+
+let analyticsHost = '';
+let pageViewId = '';
+let profileId = '';
+let sessionId = '';
+let parentId = '';
+let pageEngagement = {};
+
+let carbonAdapter = Object.assign(adapter({analyticsHost, ANALYTICS_TYPE}), {
+  track({eventType, args}) {
+    args = args ? JSON.parse(JSON.stringify(args)) : {};
+    switch (eventType) {
+      case CONSTANTS.EVENTS.AUCTION_END: {
+        registerEngagement();
+        let event = createBaseEngagementEvent(args);
+        sendEngagementEvent(event, 'auction_end');
+        break;
+      }
+      case CONSTANTS.EVENTS.TCF2_ENFORCEMENT: {
```

Comment:
This event is fired ever single time after an auction ends.

Even if consent management is not even defined or being used.

So majority of the data coming into your endpoint would look something like this:

```json
{
    "consent": {
        "gdpr_consent": "",
        "ccpa_consent": ""
    },
    "tcf_events": {
        "storageBlocked": [],
        "biddersBlocked": [],
        "analyticsBlocked": []
    }
}
```

Is this worth it?

1. Is this event needed for your system at all
2. If so, why and what do you need it for, can we add some logic to ignore this event if all of the tcf_events arrays are empty?

Reply:
I think this is a valid point, I'm going to add logic so we check if we actually have results in the tcf_events object before recording an empty result.